flask_mwoauth/__init__.py

```diff
@@ -6,94 +6,75 @@
-from flask_oauthlib.client import OAuth, OAuthException
+import mwoauth
+import requests
 from requests.models import Request
+from requests_oauthlib import OAuth1
+
+__version__ = '0.2.54'
 
 
 class MWOAuth(object):
-    def __init__(self,
+    def __init__(self, consumer_key, consumer_secret,
                  base_url='https://www.mediawiki.org/w',
-                 clean_url='https://www.mediawiki.org/wiki',
                  default_return_to='index',
-                 consumer_key=None, consumer_secret=None, name='mw.org'):
-        if not consumer_key or not consumer_secret:
-            raise Exception('MWOAuthBlueprintFactory needs consumer key and secret')
-        self.base_url = base_url
+                 name="Deprecated",
+                 clean_url="Deprecated"):
 
+        consumer_token = mwoauth.ConsumerToken(consumer_key, consumer_secret)
         self.default_return_to = default_return_to
+        self.script_url = base_url + "/index.php"
+        self.api_url = base_url + "/api.php"
 
-        self.oauth = OAuth()
-        request_url_params = {'title': 'Special:OAuth/initiate',
-                              'oauth_callback': 'oob'}
-        access_token_params = {'title': 'Special:OAuth/token'}
-        self.mwoauth = self.oauth.remote_app(
-            name,
-            base_url=base_url + "/index.php",
-            request_token_url=base_url + "/index.php?" +
-                              urlencode(request_url_params),
-            request_token_params=None,
-            access_token_url=base_url + "/index.php?" +
-                             urlencode(access_token_params),
-            authorize_url=clean_url + '/Special:OAuth/authorize',
-            consumer_key=consumer_key,
-            consumer_secret=consumer_secret,
-        )
-
-        @self.mwoauth.tokengetter
-        def get_mwo_token(token=None):
-            return session.get('mwo_token')
+        self.handshaker = mwoauth.Handshaker(self.script_url, consumer_token)
 
         self.bp = Blueprint('mwoauth', __name__)
 
         @self.bp.route('/logout')
         def logout():
-            session['mwo_token'] = None
+            session['access_token'] = None
             session['username'] = None
             if 'next' in request.args:
                 return redirect(request.args['next'])
             return "Logged out!"
 
         @self.bp.route('/login')
         def login():
-            uri_params = {'oauth_consumer_key': self.mwoauth.consumer_key}
-            redirector = self.mwoauth.authorize(**uri_params)
+            redirect_to, request_token = self.handshaker.initiate()
+            session['request_token'] = \
+                dict(zip(request_token._fields, request_token))
 
             if 'next' in request.args:
-                oauth_token = session[self.mwoauth.name + '_oauthtok'][0]
-                session[oauth_token + '_target'] = request.args['next']
+                session['next'] = request.args.get('next')
```

Comment:
The key should refer to a specific oauth login request. Otherwise, this will cause a race condition when two login requests are done in parallel. Each login request should return to their own target.